app/commands/iteration/destroy.rb

```diff
@@ -9,12 +9,11 @@ def call
 
       Solution::Uncomplete.(solution) unless solution.iterations.not_deleted.exists?
       Solution::Unpublish.(solution) if solution.published_iteration_id == iteration.id
+      Solution::UpdateNumLoc.(solution)
```

Comment:
This makes sure that when an iteration is destroyed, the num_loc of the solution is updated (could be a no-op).
The scenarios are:

1. The solution was the single published iteration: update num_loc to latest iteration's num_loc
2. The solution was the latest published iteration: update num_loc to latest published iteration's num_loc
3. The solution had one published iteration and this was not it: don't update num_loc
4. The solution was not published and this was the latest iteration: update num_loc to latest published iteration's num_loc
5. The solution was not published and this wasn't the latest iteration: don't update num_loc